src/patch.ml

```diff
@@ -219,36 +224,55 @@ let pp ~git ppf t =
   pp_operation ~git ppf t.operation ;
   List.iter (pp_hunk ppf) t.hunks
 
-let operation_of_strings git mine their =
+let get_filename ~p full =
+  let rec iter idx = function
+    | 0 -> String.sub full idx (String.length full - idx)
+    | p -> iter (String.index_from full idx '/') (p - 1)
```

Comment:
TODO: `/` and `\` should both be valid

Reply:
sure? I'm a bit uncertain how paths with spaces are escaped in a diff...

Reply:
```
--- "a b"	2024-04-02 13:32:43.427214939 +0100
+++ "a c"	2024-04-02 13:32:34.520202398 +0100
@@ -1 +0,0 @@
-test
```
with quotes apparently

Reply:
actually git diff doesn't do that:
```
diff --git a/a b b/a c
index 039727e..e69de29 100644
--- a/a b       
+++ b/a c       
@@ -1 +0,0 @@
-test
```

Reply:
and busybox doesn't do anything:
```
--- a b
+++ a c
@@ -1 +0,0 @@
-test
```

Reply:
thanks for investigating. so, dealing with `\` in paths would be nice (and doesn't interact badly with filenames containing spaces)

Reply:
fixed in #20